File: 2024/july/javaScript/08.js
/**
 * @param {number[][]} customers
 * @return {number}
 */
var averageWaitingTime = function(customers) {
    let ans = 0;
    const sz = customers.length;

        let st = 0;
        for (const [first , second] of customers){
            st = Math.max(st , first);
            const end = st + second;
            ans+= end - first;
            st = end;
        }

        return ans / sz;
};

File: 2024/july/java/08.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        double ans = 0;
        final int sz = customers.length;

        long st = 0;
        for (int []ch : customers){
            st = Math.max(st , ch[0]);
            long end = st + ch[1];
            ans+= end - ch[0];
            st = end;
        }

        return ans / sz;
        
    }
}
